chain/events/filter/index.go

```diff
@@ -510,7 +545,7 @@ func (ei *EventIndex) prefillFilter(ctx context.Context, f *eventFilter, exclude
 			if len(vals) > 0 {
```

Comment:
@aarshkshah1992 in `{Get,Subscribe}ActorEventsRaw` we advertise the `"fields"` param that gets passed in here as `keysWithCodec` like this:

```
	// Matches events with the specified key/values, or all events if empty.
	// If the value is an empty slice, the filter will match on the key only, accepting any value.
```

But I think this is incorrect—"empty slice" is `len(vals)==0` which means we never make a join for the `eeX.key=?` case. Do you agree? If so we should fix this, either by changing docs or moving the `len(vals)>0` down to the `subclauses` section in this block. It can be done in a separate PR as it needs a test.

Reply:
Yeah I dug through this and this needs to be fixed. Will raise a PR tomorrow for this. Let's keep this comment unresolved till then.

Reply:
Made it an issue so it doesn't get lost, thanks for confirming: https://github.com/filecoin-project/lotus/issues/12264